runtime/onert/core/src/util/shapeinf/Squeeze.cc

```diff
@@ -0,0 +1,109 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape inferSqueezeShape(const ir::Shape &in_shape, const ir::operation::Squeeze::Param &param)
+{
+    const int ndims = param.ndim;
+    const int* squeeze_dims = param.dims;
+    bool should_squeeze[8] = {false};
+    int num_squeezed_dims = 0;
+    if (ndims == 0) {//전체 dims이 1인것 제거
+        for (int idx = 0; idx < in_shape.rank(); ++idx)
+        {
+            if (in_shape.dim(idx) == 1) {
+                should_squeeze[idx] = true;
+                ++num_squeezed_dims;
+            }
+        }
+    } else {
+        for (int idx = 0; idx < ndims; ++idx) {
+            int current = squeeze_dims[idx];
+            if (current < 0) {
+                current = squeeze_dims[idx] + in_shape.rank();
+            } else {
+                current = squeeze_dims[idx];
+            }
+
+            if (!(current >= 0 && current < in_shape.rank() && in_shape.dim(current) == 1)) {
+                throw std::runtime_error("The following conditions must be met: 0 <= dim < Shape rank, dim == 1");
+            }
+
+            if (!should_squeeze[current]) {
+                ++num_squeezed_dims;
+            }
+            should_squeeze[current] = true;
+        }   
+    }
+
+    // Set output shape.
+    ir::Shape out_shape(in_shape.rank() - num_squeezed_dims);
+    for (int in_idx = 0, out_idx = 0; in_idx < num_squeezed_dims; ++in_idx) {
+        if (!should_squeeze[in_idx]) {
+            out_shape.dim(out_idx++) = in_shape.dim(in_idx);
+        }
+    }
+
+    return out_shape;
+}
+
+void StaticInferer::visit(const ir::operation::Squeeze &op)
+{
+    const auto input_idx{op.getInputs().at(ir::operation::Squeeze::Input::INPUT)};
+    const auto &input = _operands.at(input_idx);
+
+    const auto output_idx = op.getOutputs().at(0);
+    ir::Operand &output = _operands.at(output_idx);
+
+    if (input.info().isDynamic()) {
+        output.info().setDynamic();
+        return;
+    }
+
+    // Squeeze output shpae
+    ir::Shape new_shape = inferSqueezeShape(input.info().shape(), op.param());
+    output.info().shape(new_shape);
+}
+
+void DynamicInferer::visit(const ir::operation::Squeeze &op)
+{
+    const auto input_idx{op.getInputs().at(ir::operation::Squeeze::Input::INPUT)};
+    const auto &input = _tensor_registry->getITensor(input_idx);
+
+    if (!(input->is_dynamic())) {
+        return;
+    }
+
+    auto input_shape = getShape(input.get());
+
+    // Squeeze output shpae
+    ir::Shape new_shape = inferSqueezeShape(input.info().shape(), op.param());
```

Comment:
you already get shape of input "input_shape = getShape(input.get());"
so I think you don't need to invoke input.info().shape()